### PR TITLE
Product metadata in app router

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Fetch global product metadata (aka availableProducts) in app router

## Justify why they are needed

- Needed for generic layout
- Demonstate working with ApolloClient for React Server Components (RSC) in statically built pages

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3948?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3929** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3936** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3936?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3939** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3939?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3893** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3893?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #3923** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #3948** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3948?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3948?utm_source=stack-comment).

File: apps/store/src/services/apollo/app-router/rscClient.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+type RscClientParams =
+  | {
+      locale: RoutingLocale
+    }
+  | undefined
+
+type GetClient = (params: RscClientParams) => ApolloClient<any>
+
+// TODO: Support RSC client for dynamic pages too (auth headers, hedvig-shop-session-id header)
+export const { getClient }: { getClient: GetClient } = registerApolloClient(
```
Comment (guilhermespopolin) [inline apps/store/src/services/apollo/app-router/rscClient.ts:31]: [Suggestion]
Should we call it `getApolloClient`? I understand that based on the context it get's important one can figure that out but I think it's easier to read/understand in code right away if we were more explicitly on which client we're talking about.

**Review by guilhermespopolin — APPROVED** [review]
Way to go. Just left some minor comments.

File: apps/store/src/app/[locale]/ProductMetadataProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+type Props = {
+  productMetadata: GlobalProductMetadata
+  locale: RoutingLocale
+  children: ReactNode
+}
+// Provides jotai value for any client components in subtree
+// This needs to be separate from `useHydrateProductMetadata`:
+// - we cannot use next/router in app dir
+// - we cannot use jotai global store between server renders (https://jotai.org/docs/guides/nextjs#provider)
```
[the remark this answers is not in the document]
Comment (alebedev) [reply apps/store/src/app/[locale]/ProductMetadataProvider.tsx:16]: Right, this comment is no longer relevant

[the remark this answers is not in the document]
Comment (alebedev) [reply apps/store/src/app/[locale]/ProductMetadataProvider.tsx:16]: Refactored to use the same hydrate hooks for app and pages routers now

**Pull request merged**